packages/native-stack/src/utils/useAnimatedHeaderHeight.tsx

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+import type { Animated } from 'react-native';
+
+export const AnimatedHeaderHeightContext = React.createContext<
+  Animated.Value | undefined
+>(undefined);
+
+export function useAnimatedHeaderHeight() {
+  const animatedValue = React.useContext(AnimatedHeaderHeightContext);
+
+  if (animatedValue === undefined) {
+    throw new Error(
+      "Couldn't find the header height. Are you inside a screen in a navigator with a header?"
```

Comment:
Let's say "screen in a native stack navigator"

Reply:
Done! https://github.com/react-navigation/react-navigation/pull/11663/commits/02ad6eb0028d1bdd83ecf520eb293c20b9fc392e 👍